index.js

```diff
@@ -1040,7 +1052,7 @@ function buildArray (location, code, name, key = null) {
   switch (largeArrayMechanism) {
     case 'default':
       code += `
-      return \`[\${obj.map(${result.mapFnName}).join(',')}]\``
+      return \`[\${obj.map((value) => ${result.mapFnName}(value)).join(',')}]\``
```

Comment:
let's remove this .map() while we are at it and use the following pattern:

```
cons res = new Array(obj.length)
for (let i = 0 ; i < obj.length; i++) {
...
}
```

it's significantly faster than .map()